src/Fixer/DeprecatedFixerInterface.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer;
+
+/**
+ * @author Kuba Werłos <werlos@gmail.com>
+ */
+interface DeprecatedFixerInterface extends FixerInterface
+{
+    /**
+     * Returns name of the fixer to use instead.
+     *
+     * @return string
+     */
+    public function getSuccessor();
```

Comment:
Sound a YAGNI to me - do we consider any existing fixer to be removed without an alternative? Similarly, I can't think of any case we would split a fixer into smaller ones. We could make the interface internal, so in future we could easily implement this change if it would be necessary.

Reply:
>Sound a YAGNI to me - do we consider any existing fixer to be removed without an alternative? 

PSR0 fixer is likely to be dropped in 3.0 without an alternative.

>Similarly, I can't think of any case we would split a fixer into smaller ones. We could make the interface internal, so in future we could easily implement this change if it would be necessary.

https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2619/files#diff-88b99bb28683bd5b7e3a204826ead112R1025

this case is happening now/soon :)